projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -192,6 +194,15 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
     }
   }
 
+  private triggerControlValidation() {
+    if (this.datepickerHasFormControl()) {
+      this.control?.control?.updateValueAndValidity({ emitEvent: false }); // Added emitEvent purposefully inorder to prevent unnecessary valuechange triggers
+      if (this.control.control && this.control.control.touched) {
+        this.control.control.setErrors(this.control.control.errors); // Inorder to revalidate and apply validation errors(if any) only when form control is touched & trigger status change event.
```

Comment:
```suggestion
        // Apply validation errors (if any) and trigger status change event when control is touched.
        this.control.control.setErrors(this.control.control.errors);
```

Same here. I'd put the comment on the preceding line so that the line isn't so long.

I also reworded a bit. If I understand correctly, this line doesn't revalidate. That is done by `updateValueAndValidity`.